tracing/start_options.go

```diff
@@ -0,0 +1,54 @@
+package tracing
+
+import (
+	opentracing "github.com/opentracing/opentracing-go"
+)
+
+// StartSpanOptions is the additional options for baseplate spans.
+type StartSpanOptions struct {
+	Type          SpanType
+	ComponentName string
+}
+
+// StartSpanOption defines additional options for baseplate spans.
+type StartSpanOption interface {
+	opentracing.StartSpanOption
+
+	ApplyBP(*StartSpanOptions)
```

Comment:
A bit confused about the semantic here. Why is it saying ApplyBP? BP stands for Baseplate here?

Reply:
Yes BP stands for Baseplate. The [original API](https://pkg.go.dev/github.com/opentracing/opentracing-go?tab=doc#StartSpanOption) is limited to whatever [`opentracing.StartSpanOptions`](https://pkg.go.dev/github.com/opentracing/opentracing-go?tab=doc#StartSpanOptions) holds and that's not extensible from here, so we have to extend that in this way.